### PR TITLE
consistent energy model

### PR DESCRIPTION
**Pull request by njzjz**
Many hacking things...

### CODE REVIEW

File: deepmd/tf/model/model.py [hunk 1]
```diff
@@ -730,3 +787,65 @@ def update_sel(cls, global_jdata: dict, local_jdata: dict):
...
+            The deserialized descriptor
+        """
+        data = copy.deepcopy(data)
+
+        data["descriptor"]["type"] = {
+            "DescrptSeA": "se_e2_a",
+        }[data.pop("descriptor_name")]
+        data["fitting"]["type"] = {
+            "EnergyFittingNet": "ener",
+        }[data.pop("fitting_name")]
```
Comment (njzjz) [inline deepmd/tf/model/model.py:817]: The hack thing is the manual conversion here... It needs to be discussed.

File: source/tests/consistent/model/common.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+
+    def eval_dp_model(self, dp_obj: Any, natoms, coords, atype, box) -> Any:
+        return dp_obj(coords, atype, box=box)
+
+    def eval_pt_model(self, pt_obj: Any, natoms, coords, atype, box) -> Any:
+        return {
+            kk: vv.detach().cpu().numpy() if torch.is_tensor(vv) else vv
+            for kk, vv in pt_obj(
+                torch.from_numpy(coords).to(PT_DEVICE),
```
Comment (wanghan-iapcm) [inline source/tests/consistent/model/common.py:61]: use `pt.utils.utils.to_numpy_array` ?

File: source/tests/consistent/model/test_ener.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+    tf_class = EnergyModelTF
+    dp_class = EnergyModelDP
+    pt_class = EnergyModelPT
+    args = model_args()
+
+    def pass_data_to_cls(self, cls, data) -> Any:
+        """Pass data to the class."""
+        data = data.copy()
+        if cls is EnergyModelDP:
+            # should not do it here...
```
Comment (wanghan-iapcm) [inline source/tests/consistent/model/test_ener.py:88]: I agree... we need a `get_model` in dpmodel.

File: source/tests/consistent/model/common.py (hunk 2 above)
Comment (njzjz) [reply source/tests/consistent/model/common.py:61]: Need `pt.utils.utils.from_numpy_array` for this line, not `to_numpy_array`.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**